catalogue_of_life/Loader.java

```diff
@@ -0,0 +1,242 @@
+package com.vaticle.typedb.example.catalogueOfLife;
+
+import cli.LoadOptions;
+import com.univocity.parsers.common.record.Record;
+import com.univocity.parsers.tsv.TsvParser;
+import com.univocity.parsers.tsv.TsvParserSettings;
+import com.univocity.parsers.tsv.TsvWriter;
+import com.univocity.parsers.tsv.TsvWriterSettings;
+import loader.TypeDBLoader;
+import picocli.CommandLine;
+
+import java.io.*;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+public class Loader {
+    public static void main(String[] args) throws IOException {
+        Optional<Options> maybeOptions = parseCLIOptions(args);
+        if (maybeOptions.isEmpty()) return;
+
+        Options options = maybeOptions.get();
+
+        Path workingDirectory = Path.of(options.workingDirectory());
+        Path zipFile = fetchZip(workingDirectory, options);
+
+        Path dataDirectory = workingDirectory.resolve("coldp");
+        if (Files.exists(zipFile) && (options.forceUnzip() || !Files.exists(dataDirectory))) unzip(zipFile, dataDirectory);
+
+        prepareData(dataDirectory);
+        loadData(dataDirectory, options);
+    }
+
+    public static Optional<Options> parseCLIOptions(String[] args) {
+        CommandLine commandLine = new CommandLine(new Options());
+        try {
+            CommandLine.ParseResult parseResult = commandLine.parseArgs(args);
+            if (commandLine.isUsageHelpRequested()) {
+                commandLine.usage(commandLine.getOut());
+                return Optional.empty();
+            } else if (commandLine.isVersionHelpRequested()) {
+                commandLine.printVersionHelp(commandLine.getOut());
+                return Optional.empty();
+            } else {
+                return Optional.of(parseResult.asCommandLineList().get(0).getCommand());
+            }
+        } catch (CommandLine.ParameterException ex) {
+            commandLine.getErr().println(ex.getMessage());
+            if (!CommandLine.UnmatchedArgumentException.printSuggestions(ex, commandLine.getErr())) {
+                ex.getCommandLine().usage(commandLine.getErr());
+            }
+            return Optional.empty();
+        }
+    }
+
+    public static Path fetchZip(Path workingDirectory, Options options) throws MalformedURLException {
+        URL url = new URL("https://api.checklistbank.org/dataset/9817/export.zip?format=ColDP");
+
+        Path zipFile = workingDirectory.resolve("coldp.zip");
+        if (!options.skipDownload() && (options.forceDownload() || !Files.exists(zipFile))) {
```

Comment:
If the user did not request to skip the download, and either the file does not exist or the download is forced (but swap those last two for potential short circuit without pinging the filesystem)